src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,357 @@
+      test('renders correctly with without badge', async () => {
+        const ref = React.createRef<React.Ref<ButtonProps.Ref>>();
+        const { wrapper, getByTestId } = await renderVisualRefreshTriggerButton(
+          {
+            badge: false,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref as any
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(wrapper!.findIcon()).toBeTruthy();
+        expect(wrapper.findByClassName(visualRefreshStyles.dot)).toBeNull();
+        expect(wrapper.findBadge()).toBeNull();
+      });
+
+      test('renders correctly with aria controls', async () => {
+        const ref = React.createRef<React.Ref<ButtonProps.Ref>>();
+        const mockAriaControls = 'mock-aria-control';
+        const { wrapper, getByTestId } = await renderVisualRefreshTriggerButton(
+          {
+            badge: false,
+            ariaControls: mockAriaControls,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref as any
+        );
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(button!.getElement().getAttribute('aria-controls')).toEqual(mockAriaControls);
+      });
+
+      test('renders correctly with iconName', async () => {
+        const ref = React.createRef<React.Ref<ButtonProps.Ref>>();
+        const { wrapper, getByTestId } = await renderVisualRefreshTriggerButton({}, {}, ref as any);
+
+        expect(wrapper).not.toBeNull();
+        const button = wrapper.find('button');
+        expect(getByTestId(mockTestId)).toBeTruthy();
+        expect(button).toBeTruthy();
+        expect(button!.getElement().getAttribute('aria-label')).toEqual(mockProps.ariaLabel);
+        expect(wrapper!.findIcon()).toBeTruthy();
+      });
+
+      test('renders correctly with iconSvg', async () => {
+        const iconTestId = 'icon-test-id';
+        const icon = (
+          <svg data-testid={iconTestId} viewBox="0 0 24 24">
+            <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z" />
+          </svg>
+        );
+        const ref = React.createRef<React.Ref<ButtonProps.Ref>>();
+        const { wrapper, getByTestId } = await renderVisualRefreshTriggerButton(
+          {
+            iconSvg: icon,
+          },
+          {
+            isMobile,
+            hasOpenDrawer,
+          },
+          ref as any
```

Comment:
Why?

Reply:
`Argument of type 'RefObject<Ref<Ref>>' is not assignable to parameter of type 'Ref<Ref>'.
      Type 'RefObject<Ref<Ref>>' is not assignable to type 'RefObject<Ref>'.`
      
      the other alternative to any is to cast as `ref as unknown as React.Ref<ButtonProps.Ref>`